provider/datagen/src/lib.rs

```diff
@@ -125,10 +119,44 @@ pub fn keys<S: AsRef<str>>(strings: &[S]) -> Vec<ResourceKey> {
 /// # Ok(())
 /// # }
 /// ```
-pub fn keys_from_file<R: Read>(file: R) -> std::io::Result<Vec<ResourceKey>> {
-    let keys = BufReader::new(file)
-        .lines()
-        .collect::<std::io::Result<HashSet<String>>>()?;
+pub fn keys_from_bin<P: AsRef<Path>>(
```

Comment:
With this you can actually point your `build.rs` at your target folder and then find a fixed point in your compilation. Luckily it's just one step.